packages/owl-bot/src/owl-bot.ts

```diff
@@ -49,11 +50,23 @@ interface PubSubContext {
   };
 }
 
-export function OwlBot(
-  privateKey: string | undefined,
-  app: Probot,
-  db?: Db
-): void {
+async function acquireLock(target: string): Promise<DatastoreLock | boolean> {
+  const lock = new DatastoreLock('owlbot', target, 25 * 1000);
+  if (await lock.peek()) {
+    return false;
+  }
+  const result = await lock.acquire();
+  if (!result) {
+    return false;
+  }
+  return lock;
+}
+
+function isLock(lock: DatastoreLock | boolean): lock is DatastoreLock {
```

Comment:
Instead of returning two different types in `acquireLock` and then adding this function, what if we just returned the lock or else threw an exception and handled it?